File: DrawIt/MyGlow.vb
Imports System.ComponentModel
Imports System.Runtime.CompilerServices

<Serializable>
Public Class MyGlow
	Implements INotifyPropertyChanged, ICloneable

#Region "Enum"
	Public Enum GlowStyle
		OnShape
		OnBorder
	End Enum

	Public Enum Clip
		None
		Inside
		Outside
	End Enum

#End Region

#Region "Event"
	<NonSerialized>
	Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

	Private Sub NotifyPropertyChanged(<CallerMemberName> Optional propertyName As String = "")
		RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
	End Sub
#End Region

#Region "Properties"
	Private _enabled As Boolean = False
	Public Property Enabled() As Boolean
		Get
			Return _enabled
		End Get
		Set(value As Boolean)
			If Not value = _enabled Then
				_enabled = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private _style As GlowStyle = GlowStyle.OnShape
	Public Property GStyle() As GlowStyle
		Get
			Return _style
		End Get
		Set(value As GlowStyle)
			If Not value = _style Then
				_style = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private _clip As Clip = Clip.None
	Public Property GClip() As Clip
		Get
			Return _clip
		End Get
		Set(value As Clip)
			If Not value = _clip Then
				_clip = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private _before As Boolean = True
	Public Property BeforeFill() As Boolean
		Get
			Return _before
		End Get
		Set(value As Boolean)
			If Not value = _before Then
				_before = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private _glow As Integer = 35
	Public Property Glow() As Integer
		Get
			Return _glow
		End Get
		Set(value As Integer)
			If Not value = _glow Then
				_glow = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private _feather As Integer = 35
	Public Property Feather() As Integer
		Get
			Return _feather
		End Get
		Set(value As Integer)
			If Not value = _feather Then
				_feather = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private _clr As Color = Color.Red
	Public Property GlowColor() As Color
		Get
			Return _clr
		End Get
		Set(value As Color)
			If Not value = _clr Then
				_clr = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property
#End Region

#Region "Clone"
	''' <summary>
	''' Creates an exact copy of this <see cref="MyGlow"/> object.
	''' </summary>
	Public Function Clone() As Object Implements ICloneable.Clone
		Dim _new As New MyGlow
		For Each pd As PropertyDescriptor In TypeDescriptor.GetProperties(GetType(MyGlow))
			pd.SetValue(_new, pd.GetValue(Me))
		Next
		Return _new
	End Function
#End Region

End Class

File: DrawIt.Models/Classes/MyGlow.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace DrawIt.Models
{

	[Serializable]
	public class MyGlow : INotifyPropertyChanged, ICloneable
	{
		public event PropertyChangedEventHandler? PropertyChanged;

		private void NotifyPropertyChanged([CallerMemberName] string propertyName = "")
		{
			PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
		}

		private bool _enabled = false;
		public bool Enabled
		{
			get
			{
				return _enabled;
			}
			set
			{
				if (!value.Equals(_enabled))
				{
					_enabled = value;
					NotifyPropertyChanged();
				}
			}
		}

		private GlowStyle _style = GlowStyle.OnShape;
		public GlowStyle GStyle
		{
			get
			{
				return _style;
			}
			set
			{
				if (!value.Equals(_style))
				{
					_style = value;
					NotifyPropertyChanged();
				}
			}
		}

		private GlowClip _clip = GlowClip.None;
		public GlowClip GClip
		{
			get
			{
				return _clip;
			}
			set
			{
				if (!value.Equals(_clip))
				{
					_clip = value;
					NotifyPropertyChanged();
				}
			}
		}

		private bool _before = true;
		public bool BeforeFill
		{
			get
			{
				return _before;
			}
			set
			{
				if (!value.Equals(_before))
				{
					_before = value;
					NotifyPropertyChanged();
				}
			}
		}

		private int _rad = 35;
		public int Radius
		{
			get
			{
				return _rad;
			}
			set
			{
				if (!value.Equals(_rad))
				{
					_rad = value;
					NotifyPropertyChanged();
				}
			}
		}

		private int _strength = 35;
		public int Strength
		{
			get
			{
				return _strength;
			}
			set
			{
				if (!value.Equals(_strength))
				{
					_strength = value;
					NotifyPropertyChanged();
				}
			}
		}

		private Color _clr = Color.Red;
		public Color GlowColor
		{
			get
			{
				return _clr;
			}
			set
			{
				if (!value.Equals(_clr))
				{
					_clr = value;
					NotifyPropertyChanged();
				}
			}
		}

		/// <summary>
		///		Creates an exact copy of this <see cref="MyGlow"/> object.
		/// </summary>
		public object Clone()
		{
			MyGlow _new = new();
			foreach (PropertyDescriptor pd in TypeDescriptor.GetProperties(typeof(MyGlow)))
				pd.SetValue(_new, pd.GetValue(this));
			return _new;
		}
	}

}
